packages/gatsby/cache-dir/dev-loader.js

```diff
@@ -77,6 +77,9 @@ class DevLoader extends BaseLoader {
   }
 
   doPrefetch(pagePath) {
+    if (process.env.GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND) {
+      return Promise.resolve()
+    }
```

Comment:
We call this whenever link is in viewport - this makes dev server / runtime feel unresponsive when there is a lot of heavy pages linked.

We will continue to preload resources on hovering for now - it does have similar potential to cause unresponsiveness if someone flicks their mouse through list of links (i.e. horizontal menu bars or vertical sidebar with bunch of links), but overall hover behaviour seems safer to keep because we can start running queries just a bit sooner than if we would wait for a click making delay in loading page feel a bit faster (perceived perf is magic btw :) )